Lib/importlib/abc.py

```diff
@@ -382,23 +388,35 @@ def read_text(self, encoding=None):
     def is_dir(self) -> bool:
         """
         Return True if self is a dir
+
+        If the directory/file cannot be found, FileNotFoundError should be raised,
+        although this is not guaranteed for compatibility reasons.
         """
 
     @abc.abstractmethod
     def is_file(self) -> bool:
         """
         Return True if self is a file
+
+        If the directory/file cannot be found, FileNotFoundError should be raised,
+        although this is not guaranteed for compatibility reasons.
         """
 
     @abc.abstractmethod
     def joinpath(self, child):
         """
         Return Traversable child in self
+
+        If the file cannot be found, FileNotFoundError should be raised,
```

Comment:
This comment is incorrect. Joinpath doesn't perform any check for existence or type.